packages/redux/src/checkout/reducer.ts

```diff
@@ -172,20 +172,70 @@ const convertCheckoutOrder = (
     !!get(mergedState, `checkoutOrders[${result}].clickAndCollect`) &&
     !get(entities, `checkoutOrders[${result}].clickAndCollect`);
 
-  if (!shouldResetClickAndCollect) {
-    return mergedState;
+  const shouldResetShippingState =
+    (!!get(
+      mergedState,
+      `checkoutOrders[${result}].shippingAddress.state.code`,
+    ) ||
+      !!get(
+        mergedState,
+        `checkoutOrders[${result}].shippingAddress.state.name`,
+      )) &&
+    (!get(entities, `checkoutOrders[${result}].shippingAddress.state.code`) ||
+      !get(entities, `checkoutOrders[${result}].shippingAddress.state.name`));
+
+  const shouldResetBillingState =
+    (!!get(
+      mergedState,
+      `checkoutOrders[${result}].billingAddress.state.code`,
+    ) ||
+      !!get(
+        mergedState,
+        `checkoutOrders[${result}].billingAddress.state.name`,
+      )) &&
+    (!get(entities, `checkoutOrders[${result}].billingAddress.state.code`) ||
+      !get(entities, `checkoutOrders[${result}].billingAddress.state.name`));
+
+  let returnedState = mergedState;
+
+  if (shouldResetShippingState || shouldResetBillingState) {
+    returnedState = {
+      ...returnedState,
```

Comment:
As stated before, using immer will make these updates a lot easier and less error prone.